hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java

```diff
@@ -556,6 +569,9 @@ public void terminateDatanode() {
 
   @Override
   public void stop() {
+    if (CleanUpManager.checkContainerSchemaV3Enabled(conf)) {
+      cleanTmpDir();
+    }
```

Comment:
The shutdown hook should be moved inside the atomic boolean check to make sure it is only invoked once.